libnd4j/include/ops/declarable/generic/parity_ops/slice.cpp

```diff
@@ -37,13 +37,19 @@ namespace nd4j {
 
             IndicesList indices;
             for (int e = 0; e < x_rank; e++) {
-                int stop = end[e];
+                int size = end[e];
                 int start = begin[e];
 
+                REQUIRE_TRUE(start >= 0, 0, "Slice: start index should not be negative");
 
-                REQUIRE_TRUE(stop > 0, 0, "Slice: interval for dimension %i is less then 1", e);
+                REQUIRE_TRUE(start < input->shapeInfo()[e + 1], 0, "Index %i is invalid for dimension %i with size %i.", start, e, input->shapeInfo()[e + 1]);
+                if (size == -1){
+                    size = input->shapeInfo()[e + 1] - start;
+                }
+                REQUIRE_TRUE(size > 0, 0, "Slice: interval for dimension %i is less then 1");
+                REQUIRE_TRUE(start + size <= input->shapeInfo()[e + 1], 0, "Slice: interval [%i, %i] is out of bounds for dimension %i with size %i", start, start + size, e, input->shapeInfo()[e + 1]);
```

Comment:
Conditions look correct. Only thing is whether ->sizeOf(e) instead of shapeInfo()[e+1] would be better/clearer here?